client/fingerprint_manager.go

```diff
@@ -113,6 +141,23 @@ func (fm *FingerprintManager) fingerprint(name string, f fingerprint.Fingerprint
 	return response.Detected, nil
 }
 
+// healthcheck checks the health of the specified resource.
+func (fm *FingerprintManager) healthCheck(name string, hc fingerprint.HealthCheck) error {
+	request := &cstructs.HealthCheckRequest{}
+	var response cstructs.HealthCheckResponse
+	if err := hc.Check(request, &response); err != nil {
+		return err
+	}
+
```

Comment:
If the value of the health check hasn't changed avoid calling `updateHealthCheck`. Otherwise every minute we will be causing a the client to push changes to the server and cause unnecessary Raft transactions.

Reply:
I was thinking about handling that in `updateHealthCheck` itself, as #3873 moves this logic to `updateNodeFromFingeprint`